src/core/util/sys_vars.cpp

```diff
@@ -1734,35 +1739,20 @@ void mce_sys_var::get_env_params()
         neigh_uc_arp_quata = (uint32_t)atoi(env_ptr);
     }
 
-    if ((getenv(SYS_VAR_HUGETBL)) != NULL) {
-        vlog_printf(VLOG_WARNING,
-                    "******************************************************************************"
-                    "****************************************\n");
-        vlog_printf(VLOG_WARNING,
-                    "The '%s' parameter is no longer supported, please refer to '%s' in README.txt "
-                    "for more info\n",
-                    SYS_VAR_HUGETBL, SYS_VAR_MEM_ALLOC_TYPE);
-        vlog_printf(VLOG_WARNING,
-                    "******************************************************************************"
-                    "****************************************\n");
-    }
-
     if ((env_ptr = getenv(SYS_VAR_MEM_ALLOC_TYPE)) != NULL) {
-        mem_alloc_type = (alloc_mode_t)atoi(env_ptr);
-    }
-    if (mem_alloc_type < 0 || mem_alloc_type >= ALLOC_TYPE_LAST_ALLOWED_TO_USE) {
-        mem_alloc_type = MCE_DEFAULT_MEM_ALLOC_TYPE;
+        mem_alloc_type = option_alloc_type::from_str(env_ptr, MCE_DEFAULT_MEM_ALLOC_TYPE);
     }
-    if (mce_sys_var::HYPER_MSHV == hypervisor && mem_alloc_type == ALLOC_TYPE_CONTIG) {
-        char mem_str[sizeof(int) + 1] = {0};
-        len = snprintf(mem_str, sizeof(mem_str), "%d", ALLOC_TYPE_HUGEPAGES);
-        if (likely((0 < len) && (len < (int)sizeof(mem_str)))) {
-            setenv(SYS_VAR_MEM_ALLOC_TYPE, mem_str,
-                   1); // Setenv to avoid core dump while valgrind is used.
+    if ((env_ptr = getenv(SYS_VAR_HUGEPAGE_LOG2)) != NULL) {
+        unsigned val = (unsigned)atoi(env_ptr);
+
+        // mmap() uses 6 bits for the hugepage size log2
+        if (val < 64U) {
+            hugepage_log2 = val;
```

Comment:
Do we want to check if it is a supported size? otherwise use default?

Reply:
Good idea, will implement

Reply:
If requested hugepage (including non-zero `MCE_DEFAULT_HUGEPAGE_LOG2`) is not supported, XLIO will fallback to 0 (autodetection)